target/linux/ath79/dts/ar9344_comfast_cf-e120a-v3.dts

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9344.dtsi"
+
+/ {
+	compatible = "comfast,cf-e120a-v3", "qca,ar9344";
+	model = "COMFAST CF-E120A v3";
+
+	aliases {
+		serial0 = &uart;
+		led-boot = &wan;
+		led-failsafe = &wan;
+		led-upgrade = &wan;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&led_rssimediumhigh_pin>;
+
+		wan: wan {
+			label = "cf-e120a-v3:green:wan";
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "cf-e120a-v3:green:lan";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		};
+
+		rssilow {
+			label = "cf-e120a-v3:red:rssilow";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		rssimediumlow {
+			label = "cf-e120a-v3:red:rssimediumlow";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		rssimediumhigh {
+			label = "cf-e120a-v3:green:rssimediumhigh";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		rssihigh {
+			label = "cf-e120a-v3:green:rssihigh";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	ath9k-leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "cf-e120a-v3:green:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pinmux {
+		led_rssimediumhigh_pin: pinmux_rssimediumhigh_pin {
+			pinctrl-single,bits = <0x10 0x0 0xff>;
+		};
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			uboot:	partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x010000>;
+				read-only;
+			};
+
+			art: partition@10000 {
+				label = "art";
+				reg = <0x010000 0x010000>;
+				read-only;
+			};
+
+			firmware: partition@20000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x020000 0x7d0000>;
+			};
+
+			nvram: partition@7f0000 {
+				label = "nvram";
+				reg = <0x7f0000 0x010000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&eth0 {
+	status = "okay";
+	phy-handle = <&swphy0>;
+	mtd-mac-address = <&art 0x0>;
+
+	gmac-config {
+		device = <&gmac>;
+		switch-phy-swap = <1>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+	mtd-mac-address = <&art 0x1002>;
+};
+
+&wmac {
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x6>;
+};
```

Comment:
0x0 is LAN's mac, 0x6 is for wan, 0x1002 is for wlan.

Reply:
If there is a correct wlan MAC address at 0x1002, the mtd-mac-address can be removed.
mtd-mac-address is used when manufacturers are too lazy to put a correct mac address in their wlan calibration data.

Reply:
> 0x0 is LAN's mac, 0x6 is for wan, 0x1002 is for wlan.

@fall513 ,

I understand you suggest:
MAC in 0x0 ends in "B0" => lan
MAC in 0x6 ends in "B2" => wan
MAC in 0x1002 ends in "B1" => wlan

However, in the **stock** firmware, I had:
MAC ending in "B0" => lan
MAC ending in "B1" => wan
MAC ending in "B2" => wlan

It made more sense to me to keep the MACs in the same order as the stock firmware (B0=>lan, B1=> wan, B2=>wlan), but if you think the assignation you suggested (B0=>lan, B1=>wlan, B2=>wan) is better, let me know and I'll change it to that.

Reply:
> If there is a correct wlan MAC address at 0x1002, the mtd-mac-address can be removed.
> mtd-mac-address is used when manufacturers are too lazy to put a correct mac address in their wlan calibration data.

@981213 ,
There is a valid MAC address in the calibration data, indeed. I added the "mtd-mac-address" parameter to be able to keep the same MAC=>interface assignations as in the stock firmware. May I ask your opinion about the https://github.com/openwrt/openwrt/pull/1779#discussion_r255465200 above?

Thanks!

Reply:
@rogerpueyo In my opnion, it had better assign wlan mac with offset 0x1002 in the calibration data.
If it is a valid mac address, it should calculte a new mac address base on lan mac or wan mac. Anyway, I don't think it is a good way to assign lan/wan mac with contents of calibration data , because calibration data's contents are work for wifi.

Reply:
OK. I've then left the MAC in the calibration data for WLAN and the one at 0x6 for WAN:

- MAC in 0x0 ends in "B0" => lan
- MAC in 0x6 ends in "B2" => wan
- MAC in 0x1002 (caldata) ends in "B1" => wlan

Thanks for your review, @fall513 , @981213 .